Documentation/ceph-csi-troubleshooting.md

```diff
@@ -0,0 +1,388 @@
+sh-4.4# ps -ef |grep ceph
+root           1       0  0 Jan20 ?        00:00:26 /usr/local/bin/cephcsi --nodeid=minikube --type=cephfs --endpoint=unix:///csi/csi.sock --v=0 --nodeserver=true --drivername=rook-ceph.cephfs.csi.ceph.com --pidlimit=-1 --metricsport=9091 --forcecephkernelclient=true --metricspath=/metrics --enablegrpcmetrics=true
+root          69      60  0 11:55 pts/0    00:00:00 grep ceph
+```
+
+If any commands are stuck check the **dmesg** logs from the node.
+Restarting the `csi-cephfsplugin` pod may also help sometimes.
+
+If you don’t see any stuck messages, confirm the network connectivity, Ceph health, and slow ops.
+
+## RBD Stale operations
+
+Check for any stale `map/mkfs/mount` commands on the `csi-rbdplugin-xxxx` pod on the node where your application pod is scheduled.
+
+You need to exec in the `csi-rbdplugin-xxxx` pod and grep for stale operators like (`rbd map, rbd unmap, mkfs, mount` and `umount`).
+
+Identify the `csi-rbdplugin-xxxx` pod running on the node where your application is scheduled with
+`kubectl get po -o wide` and match the node names.
+
+```console
+$ kubectl exec -it csi-rbdplugin-vh8d5 -c csi-rbdplugin -- sh
+# ps -ef |grep map
+root     1297024 1296907  0 12:00 pts/0    00:00:00 grep map
+
+# ps -ef |grep mount
+root        1824       1  0 Jan19 ?        00:00:00 /usr/sbin/rpc.mountd
+ceph     1041020 1040955  1 07:11 ?        00:03:43 ceph-mgr --fsid=ba41ac93-3b55-4f32-9e06-d3d8c6ff7334 --keyring=/etc/ceph/keyring-store/keyring --log-to-stderr=true --err-to-stderr=true --mon-cluster-log-to-stderr=true --log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false --mon-host=[v2:10.111.136.166:3300,v1:10.111.136.166:6789] --mon-initial-members=a --id=a --setuser=ceph --setgroup=ceph --client-mount-uid=0 --client-mount-gid=0 --foreground --public-addr=172.17.0.6
+root     1297115 1296907  0 12:00 pts/0    00:00:00 grep mount
+
+# ps -ef |grep mkfs
+root     1297291 1296907  0 12:00 pts/0    00:00:00 grep mkfs
+
+# ps -ef |grep umount
+root     1298500 1296907  0 12:01 pts/0    00:00:00 grep umount
+
+# ps -ef |grep unmap
+root     1298578 1296907  0 12:01 pts/0    00:00:00 grep unmap
+```
+
+If any commands are stuck check the **dmesg** logs from the node.
+Restarting the `csi-rbdplugin` pod also may help sometimes.
+
+If you don’t see any stuck messages, confirm the network connectivity, Ceph health, and slow ops.
+
+## dmesg logs
+
+Check the dmesg logs on the node where pvc mounting is failing or the `csi-rbdplugin` container of the
+`csi-rbdplugin-xxxx` pod on that node.
+
+```console
+$ dmesg
+```
+
+## RBD Commands
+
+If nothing else helps, get the last executed command from the ceph-csi pod logs and run it manually inside
+the provisioner or plugin pod to see if there are errors returned even if they couldn't be seen in the logs.
+
+```console
+$ rbd ls --id=csi-rbd-node -m=10.111.136.166:6789 --key=AQDpIQhg+v83EhAAgLboWIbl+FL/nThJzoI3Fg==
```

Comment:
```suggestion
rbd ls --id=csi-rbd-node -m=10.111.136.166:6789 --key=AQDpIQhg+v83EhAAgLboWIbl+FL/nThJzoI3Fg==
```

Reply:
@leseb thanks for the review. Hopefully, I have addressed all the comments:smile: